pyaedt/modeler/cad/Modeler.py

```diff
@@ -1134,6 +1207,388 @@ def _attributes(self):
+        return self._ref_cs
+
+    @ref_cs.setter
+    def ref_cs(self, value):
+        if settings.aedt_version <= "2022.2":
+            self._ref_cs = value
+            self.update()
+        obj1 = self._modeler.oeditor.GetChildObject(self.name)
+        try:
+            obj1.SetPropValue("Reference CS", value)
+            self._ref_cs = value
+        except:
+            self._modeler.logger.error("Failed to set Coordinate CS Reference.")
+
+    @property
+    def props(self):
+        """Properties of the coordinate system.
+
+        Returns
+        -------
+        :class:`pyaedt.modeler.Modeler.CSProps`
+        """
+        if self._props or settings.aedt_version <= "2022.2" or self.name is None:
+            return self._props
+        self._get_coordinates_data()
+        return self._props
+
+    @property
+    def _part_name(self):
+        """Internally get the part name of the object where the coordinate system lays"""
+        if not self.entity_id:
+            # face_id has not been defined yet
+            return None
+        for obj in self._modeler.objects.values():
+            if obj.id == self.entity_id:
+                return obj.name
+            for face in obj.faces:
+                if face.id == self.entity_id:
+                    return obj.name
+                for edge in face.edges:
+                    if edge.id == self.entity_id:
+                        return obj.name
+                    for vertex in edge.vertices:
+                        if vertex.id == self.entity_id:
+                            return obj.name
+        return None  # part has not been found
+
+    @property
+    def _object_parameters(self):
+        """Internally named array with parameters of the object coordinate system."""
+        arg = ["Name:ObjectCSParameters"]
+        _dict2arg(self.props, arg)
+        return arg
+
+    @property
+    def _attributes(self):
+        """Internal named array for attributes of the object coordinate system."""
+        coordinateSystemAttributes = ["NAME:Attributes", "Name:=", self.name, "PartName:=", self._part_name]
+        return coordinateSystemAttributes
+
+    @pyaedt_function_handler()
+    def create(
+        self,
+        cs_type,
+        entity,
+        x_axis_entity=None,
+        move_to_end=True,
+        reverse_x_axis=False,
+        reverse_y_axis=False,
+        # y_axis_x_direction=0,
+        # y_axis_y_direction=0,
+        # y_axis_z_direction=0
+    ):
+        """Create an object coordinate system.
+
+        Parameters
+        ----------
+        cs_type : str
+            Type of Object Coordinate System.
+            Possible types are ``Offset``, ``Rotated``, ``Both``.
+        entity : int, VertexPrimitive, EdgePrimitive, FacePrimitive
+            Entity where the Object Coordinate System is anchored.
+            It can be the Id of the object or an object of type VertexPrimitive, EdgePrimitive, FacePrimitive.
```

Comment:
```suggestion
            Entity where the object coordinate system is anchored.
            You can specify the ID of the object or an object of type VertexPrimitive, EdgePrimitive,
            or FacePrimitive.
```
Will the types be links or will they need to be tagged as code?